fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsApplicationProcessWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -86,470 +93,715 @@
+
+          CommandProcessingResult result = null;
+
+          JsonArray gsimApplications = command.arrayOfParameterNamed("clientArray");
+
+          for (JsonElement gsimApplication : gsimApplications) {
+               // result=submitApplication(JsonCommand.fromExistingCommand(command,
+               // gsimApplication));
+               result = submitApplication(JsonCommand.fromExistingCommand(command, gsimApplication,
+                         gsimApplication.getAsJsonObject().get("clientId").getAsLong()));
+          }
+
+          return result;
+     }
+
+     @Transactional
+     @Override
+     public CommandProcessingResult submitApplication(final JsonCommand command) {
+          try {
+               this.savingsAccountDataValidator.validateForSubmit(command.json());
+               final AppUser submittedBy = this.context.authenticatedUser();
+
+               final SavingsAccount account = this.savingAccountAssembler.assembleFrom(command, submittedBy);
+               this.savingAccountRepository.save(account);
+               String accountNumber = "";
+               GroupSavingsIndividualMonitoring gsimAccount = null;
+               BigDecimal applicationId = BigDecimal.ZERO;
+               Boolean isLastChildApplication = false;
+
+               // gsim
+               if (account.isAccountNumberRequiresAutoGeneration()) {
+
+                    final AccountNumberFormat accountNumberFormat = this.accountNumberFormatRepository
+                              .findByAccountType(EntityAccountType.SAVINGS);
+                    // if application is of GSIM type
+                    if (account.getAccountTypes() == 5) {
+                         final Long groupId = command.longValueOfParameterNamed("groupId");
+                         // GSIM specific parameters
+                         if (command.bigDecimalValueOfParameterNamedDefaultToNullIfZero("applicationId") != null) {
+                              applicationId = command.bigDecimalValueOfParameterNamedDefaultToNullIfZero("applicationId");
+                         }
+
+                         if (command.booleanObjectValueOfParameterNamed("lastApplication") != null) {
+                              isLastChildApplication = command.booleanPrimitiveValueOfParameterNamed("lastApplication");
+                         }
+
+                         Group group = this.groupRepositoryWrapper.findOneWithNotFoundDetection(groupId);
+
+                         if (command.booleanObjectValueOfParameterNamed("isParentAccount") != null)
+                         {
+                                   // empty table check
+                              if (gsimRepository.count() != 0) {
+                                   // Parent-Not an empty table
+
+                                   accountNumber = this.accountNumberGenerator.generate(account, accountNumberFormat);
+                                   account.updateAccountNo(accountNumber + "-1");
```

Comment:
omit hyphens in account numbers.

Reply:
I agree with @nikpawar on removal of hyphens

Reply:
@nikpawar89 
Without hyphens the user can easily generate BBANs, issue check books to account holders 
standard account numbers should be 10 digits

Reply:
Yes, hyphens have been removed.